scripts/tests/test_zkai_d128_snark_receipt_timing_setup_gate.py

```
@@ -0,0 +1,193 @@
+        self.assertEqual(payload["timing_metrics"]["sample_count"], 5)
+        self.assertEqual(payload["timing_metrics"]["proof_generation_time_ms_median"], 3.0)
+        self.assertEqual(payload["timing_metrics"]["verifier_time_ms_median"], 8.0)
+        self.assertTrue(payload["all_mutations_rejected"])
+        self.assertEqual(payload["case_count"], len(GATE.EXPECTED_MUTATION_INVENTORY))
+
+    def test_rejects_source_receipt_relabeling(self) -> None:
+        payload = self.payload()
+        core = GATE._core_payload(payload)
+        core["source_receipt"]["claim_boundary"] = "RECURSION_PROVEN"
+
+        with self.assertRaisesRegex(GATE.D128SnarkTimingSetupError, "source receipt mismatch") as err:
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)